src/traits/mutable.rs

```diff
@@ -16,3 +18,10 @@ pub trait MutMatrix {
     unsafe fn swap(&mut self, idx1: usize, idx2: usize, shape: Shape);
     unsafe fn swap_with_perm(&mut self, p: &Vec<(usize, usize)>, shape: Shape);
 }
+
+pub trait ComplexMutMatrix {
```

Comment:
Note: In next PRs, this can be removed and the handling added to `pub trait MutMatrix` (with a generic)